autogpt/autogpt/agents/agent.py

```diff
@@ -116,27 +120,33 @@ def __init__(
         self.commands: list[Command] = []
 
         # Components
-        self.system = SystemComponent(legacy_config, settings.ai_profile)
+        self.system = SystemComponent(settings.ai_profile)
         self.history = ActionHistoryComponent(
             settings.history,
-            self.send_token_limit,
             lambda x: self.llm_provider.count_tokens(x, self.llm.name),
-            legacy_config,
+            legacy_config.fast_llm,
             llm_provider,
+            ActionHistoryConfiguration(max_tokens=self.send_token_limit),
```

Comment:
Not necessarily for this PR, but I think we should start using kwargs more for stuff like this.

Reply:
Acknowledged